interpreter/script/js.ml

```diff
@@ -52,21 +57,36 @@ function module(bytes, valid = true) {
   return new WebAssembly.Module(buffer);
 }
 
-function instance(bytes, imports = registry) {
+function anonInstance(bytes, imports = registry) {
   return new WebAssembly.Instance(module(bytes), imports);
 }
 
-function call(instance, name, args) {
+function instance(names, bytes, imports = registry) {
```

Comment:
It would feel cleaner to me if the harness did not to mix creation of the instance with storing it under a name. Simply introduce a separate function, say, `setInstance`, and leave all code as before except replacing the generation of `let` with the generation of a call to this function. Similarly, move calls to `getInstance` to the generated code instead of coupling them with the call/get actions. Such a separation makes it easier to customise the harness, I think.

In addition, you could simply make `instances` an object, then you don't need the get/setInstance helpers but just use `instances["name"]` where it used `name` before.

Reply:
It's coupled so the name of the instance is associated with the module bytes / module creation, since I want to send the module associated with the name, not the instance.

Another alternative is to store the module on the instance object so it can be accessed from `setInstance`, I suppose.